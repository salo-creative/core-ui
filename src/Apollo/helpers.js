import { get, hasIn, forEach, isEmpty, find } from 'lodash';

// HELPERS & CONSTANTS
export const isLoading = ({ networkStatus, skip = false }) => !skip && (
  // Initial load
  (networkStatus === 1)
  // Reload due to variables change
  || (networkStatus === 2)
  // Reload due to refetch
  || (networkStatus === 4)
);

export const isError = ({ networkStatus, error, skip = false }) => (
  error || (!skip && (networkStatus === 8)) // Network error (exclude skip as that also causes status of 8)
);

export const getRequestStatus = ({ networkStatus, error, skip = false }) => {
  return {
    requestLoading: isLoading({ networkStatus, skip }),
    requestError: isError({ networkStatus, error, skip })
  };
};

export const parseApolloError = ({ error, propName = 'APOLLO_ERROR' }) => {
  let code = 400;
  let message = 'Something went wrong!';
  const errors = [];
  const extra = {};
  let name = propName; // For custom sentry logging
  if (error.graphQLErrors) {
    // Find custom errors if they exist
    const customError = find(error.graphQLErrors, { message: 'SC_ERROR' });

    if (hasIn(customError, 'path')) {
      extra.path = customError.path.join(' -> ');
      name = propName === 'APOLLO_ERROR' ? `APOLLO_ERROR: ${ customError.path[0] }` : propName;
    }

    code = get(customError, 'extensions.status', code);

    // Handle custom error message
    const serverMessage = get(customError, 'extensions.message');
    if (serverMessage && typeof serverMessage === 'string') {
      message = serverMessage;
    } else if (typeof serverMessage === 'object') {
      message = serverMessage.message; // eslint-disable-line
      code = serverMessage.name === 'ValidationError' ? 422 : 400;
    }

    // Handle custom errors array
    const serverErrorsMessages = get(customError, 'extensions.errors');
    if (!isEmpty(serverErrorsMessages)) {
      forEach(serverErrorsMessages, errMessage => {
        if (typeof errMessage === 'string') errors.push(errMessage);
      });
    }
  }

  return {
    code,
    errors,
    extra,
    message,
    name
  };
};