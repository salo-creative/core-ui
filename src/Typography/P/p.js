import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import FluidType from '../../helpers/fluidType';

const StyledP = styled.p`
  font-weight: 400;
  width: 100%;
  color: ${ ({ color, theme }) => color || theme.font };
  text-align: ${ ({ align }) => align };
  margin: ${ ({ margin }) => margin };
  padding: ${ ({ padding }) => padding };

  ${ ({ fontSize, lineHeight, minFont, maxFont, minLine, maxLine }) => {
    if (fontSize === '' && lineHeight === '') {
      return `${ FluidType({ fontSize: [minFont, maxFont], lineHeight: [minLine, maxLine] }) }`;
    }
    return `
      font-size: ${ fontSize };
      line-height: ${ lineHeight };
    `;
  } }

  em,
  i {
      font-style: italic;
  }

  b,
  strong {
      font-weight: bold;
  }

  a {
    font-size: 1em;
    color: currentColor;
    transition: color 0.3s;
    font-weight: 600;

    &:hover {
      transition: color 0.3s;
      color: #8f8f8f;
    }
  }
  pre {
    font-family: "Courier 10 Pitch", Courier, monospace;
    font-size: 95%;
    line-height: 140%;
    white-space: pre;
    white-space: -moz-pre-wrap;
    white-space: -o-pre-wrap;
    white-space: pre-wrap;
  }

  code {
    font-family: Monaco, Consolas, "Andale Mono", "DejaVu Sans Mono", monospace;
    font-size: 95%;
    line-height: 140%;
    white-space: pre;
    white-space: -moz-pre-wrap;
    white-space: -o-pre-wrap;
    white-space: pre-wrap;
    background: #faf8f0;
  }
`;

const P = ({ children, ...props }) => {
  return <StyledP { ...props }>{ children }</StyledP>;
};

P.propTypes = {
  color: PropTypes.string,
  align: PropTypes.string,
  fontSize: PropTypes.string,
  lineHeight: PropTypes.string,
  margin: PropTypes.string,
  padding: PropTypes.string,
  minFont: PropTypes.number,
  minLine: PropTypes.number,
  maxFont: PropTypes.number,
  maxLine: PropTypes.number
};

P.defaultProps = {
  color: '',
  align: 'left',
  fontSize: '',
  lineHeight: '',
  margin: '0 0 20px',
  padding: '0',
  minFont: 14,
  maxFont: 17,
  minLine: 17,
  maxLine: 29
};

P.displayName = 'P';

export default P;