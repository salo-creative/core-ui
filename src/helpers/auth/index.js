import { isEmpty, get } from 'lodash';
import Cookies from 'universal-cookie';

const cookies = new Cookies();

/**
 * GET USER COOKIES
 * get the session cookies from the cookies available
 */
export const getSessionCookies = () => {
  let tokens = null;
  const jwt = cookies.get('SCSession');
  if (!isEmpty(jwt)) {
    tokens = { jwt };
  }
  return tokens;
};

/**
 * TOKEN EXPIRED
 * Check if the jwt has expired (expects a time string NOT THE JWT)
 */
export const tokenExpired = (expiry) => {
  if (!expiry) return false;
  const timeDiff = Math.round((Date.now() - expiry) / 1000);
  return timeDiff >= 7 * 24 * 60 * 60;
};

/**
 * GET JWT
 * Get the jwt token for the local cookies
 */
export const getJwt = () => {
  const jwt = cookies.get('SCSession');
  if (jwt && jwt.ts && !tokenExpired(jwt.ts)) {
    return get(jwt, 't', '');
  }
  // No cookie found or malformed cookie.
  return '';
};