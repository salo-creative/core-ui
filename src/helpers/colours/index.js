import { darken } from 'polished';

/**
 * COLOUR DEFINITIONS
 *
 * Fixed colour definitions to ensure
 * centralised definition of consts
 */
export const colours = {
  white: '#fff',
  black: '#00222b',
  blue: '#56c3f0',
  green: '#78d731',
  grey: '#e9ecef',
  darkGrey: '#6c757d',
  navy: '#00272e',
  orange: '#f9200',
  paleGrey: '#fafcff',
  red: '#ff190f',
  yellow: '#ffcc00'
};

/**
 * BOX SHADOW
 *
 * Centralised definition for shadows
 * @param {*} size - string value for the size. [default | medium, | large]
 * @returns box shadow css property string
 */
export const boxShadow = (size) => {
  switch (size) {
    case 'large':
    case 'small':
    case 'default':
    default:
      return '2px 2px 2px 0px rgba(0,0,0,0.2)';
  }
};

/**
 * BUTTON THEMES
 *
 * Definition for the button themes from the predefined colours
 */
export const buttonThemes = {
  white: {
    border: '#fff',
    background: '#fff',
    color: colours.black,
    borderHov: colours.paleGrey,
    backgroundHov: colours.paleGrey,
    colorHov: colours.black,
    shadow: true
  },
  clear: {
    border: colours.black,
    background: 'transparent',
    color: colours.black,
    borderHov: colours.black,
    backgroundHov: 'rgba(0,0,0,0.1)',
    colorHov: colours.black,
    shadow: false
  },
  clear_inverse: {
    border: '#fff',
    background: 'transparent',
    color: '#fff',
    borderHov: darken(0.3, '#fff'),
    backgroundHov: 'transparent',
    colorHov: darken(0.3, '#fff'),
    shadow: false
  },
  text: {
    border: 'transparent',
    background: 'transparent',
    color: colours.darkGrey,
    borderHov: 'transparent',
    backgroundHov: 'transparent',
    colorHov: colours.black,
    shadow: false
  },
  text_inverse: {
    border: 'transparent',
    background: 'transparent',
    color: colours.paleGrey,
    borderHov: 'transparent',
    backgroundHov: 'transparent',
    colorHov: '#fff',
    shadow: false
  },
  primary: {
    border: colours.blue,
    background: colours.blue,
    color: '#fff',
    borderHov: darken(0.15, colours.blue),
    backgroundHov: darken(0.15, colours.blue),
    colorHov: '#fff',
    shadow: true
  },
  secondary: {
    border: colours.navy,
    background: colours.navy,
    color: '#fff',
    borderHov: darken(0.15, colours.navy),
    backgroundHov: darken(0.15, colours.navy),
    colorHov: '#fff',
    shadow: true
  },
  error: {
    border: colours.red,
    background: colours.red,
    color: '#fff',
    borderHov: darken(0.15, colours.red),
    backgroundHov: darken(0.15, colours.red),
    colorHov: '#fff',
    shadow: true
  },
  success: {
    border: colours.green,
    background: colours.green,
    color: '#fff',
    borderHov: darken(0.15, colours.green),
    backgroundHov: darken(0.15, colours.green),
    colorHov: '#fff',
    shadow: true
  }
};

export const avatarThemes = {
  purple: {
    background: '#582A72',
    color: colours.white
  }
};