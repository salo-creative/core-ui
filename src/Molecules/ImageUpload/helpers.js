import { dataURItoBlob } from '../../helpers/files';

export const onImageDrop = async ({
  files,
  maxFileSize,
  onUpload,
  rejected,
  setFileSizeError
}) => {
  if (rejected.length > 0) {
    if (rejected[0].size >= maxFileSize) {
      setFileSizeError(true);
    }
    return;
  }
  
  if (!files.length) return;
  
  setFileSizeError(false);
  
  const file = files[0];
  const img = new Image();
  
  img.onload = (e) => {
    const sizes = {
      width: e.target.width,
      height: e.target.height
    };
  
    const reader = new FileReader();
  
    reader.onload = (event) => {
      // Image service requires a binary, not base64 so we need to convert back
      const fileBlob = dataURItoBlob(event.target.result);
      onUpload({
        blob: fileBlob, base64: event.target.result, sizes
      });
    };
    reader.readAsDataURL(file);
  };
  
  img.src = URL.createObjectURL(file);
};