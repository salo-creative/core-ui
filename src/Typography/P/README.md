# Paragraph

Paragraph is a pure styled component render. Renders a standard `<p>` with themed styling

-------

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import { P } from '@salo/core-ui/Typography/p';
```

Implement as follows

```javascript
<P>This is a title</P>
```

### For full prop types and usage see storybook info/knobs