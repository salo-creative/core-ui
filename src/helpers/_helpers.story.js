import breakpoints from './breakpoints/_breakpoints.story';
import colours from './colours/_colours.story';
import environments from './environments/_environments.story';