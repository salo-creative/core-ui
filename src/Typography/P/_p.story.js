import React from 'react';
import { storiesOf } from '@storybook/react';

// load tests
import { withTests } from '@storybook/addon-jest';
import { withKnobs, text, select, color } from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import { P } from '../../index';

// README //
import README from './README.md';

// Start of story logic
const stories = storiesOf('Atoms | Typography/P', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests({ results }));
stories.addParameters({ jest: ['p'] });

stories.add(
  'Basic',
  (() => {
    const align = select('Align', {
      'Left': 'left',
      'Right': 'right',
      'Center': 'center'
    }, 'left');
    const fontSize = text('Font size', '');
    const lineHeight = text('Line height', '');
    const colour = color('Colour', '#262729');
    const margin = text('Margin', '0 0 20px');
    return (
      <React.Fragment>
        <P
          color={ colour }
          fontSize={ fontSize }
          lineHeight={ lineHeight }
          align={ align }
          margin={ margin }
        >
          <strong>Some bold text with a break after it</strong>
          <br />consectetur adipiscing elit. Donec molestie eleifend malesuada. <em>Ohhh italic! Donec vel massa nunc.</em> Phasellus molestie mollis lacus, pretium facilisis augue tempus at. Phasellus ullamcorper interdum posuere. Proin at suscipit mauris. Aenean convallis faucibus consectetur. Nunc vel ipsum ut libero elementum sollicitudin ut sit amet ex. Nulla sagittis at risus vel bibendum. Pellentesque condimentum non nulla suscipit pellentesque. Curabitur eget purus sem. Morbi vestibulum egestas elit ac scelerisque. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nullam dapibus nunc sollicitudin ultricies aliquet. Etiam semper massa vulputate enim ornare lacinia. Curabitur fermentum venenatis turpis a hendrerit. Donec porta tellus in diam aliquet viverra.
        </P>

        <P
          color={ colour }
          fontSize={ fontSize }
          lineHeight={ lineHeight }
          align={ align }
          margin={ margin }
        >
          <strong>pre</strong> tag
          <br />
          <pre>Vivamus sodales sit amet turpis id mattis. Pellentesque eu turpis rhoncus, tempus ipsum ac, aliquet urna. Fusce imperdiet quam quis congue semper. Nunc fringilla elit enim, euismod efficitur enim iaculis id.</pre>
        </P>

        <P
          color={ colour }
          fontSize={ fontSize }
          lineHeight={ lineHeight }
          align={ align }
          margin={ margin }
        >
          <strong>code</strong> tag
          <br />
          <code>Vivamus sodales sit amet turpis id mattis. Pellentesque eu turpis rhoncus, tempus ipsum ac, aliquet urna. Fusce imperdiet quam quis congue semper. Nunc fringilla elit enim, euismod efficitur enim iaculis id.</code>
        </P>
        
        <P
          color={ colour }
          fontSize={ fontSize }
          lineHeight={ lineHeight }
          align={ align }
          margin={ margin }
        >
          Nunc efficitur dolor sed venenatis condimentum. Curabitur pretium porta dolor quis consectetur. Cras non erat et neque egestas euismod in quis tellus. Nulla facilisi. Sed consectetur vulputate nibh quis finibus. Maecenas consectetur, nibh non volutpat pretium, ligula metus commodo arcu, vel mattis sapien arcu quis nunc. Aenean scelerisque sapien eget quam accumsan, et porttitor quam ultricies. Aenean sed ante tempor, laoreet sapien vitae, eleifend lectus. Phasellus eget finibus felis, et venenatis nisl. Etiam efficitur, enim et varius volutpat, augue tortor scelerisque turpis, vel scelerisque quam odio ut lectus. Morbi mollis fringilla tortor non auctor.
        </P>
      </React.Fragment>
    );
  }), { notes: README }
);