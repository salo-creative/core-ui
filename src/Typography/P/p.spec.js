import React from 'react';
import 'jest-styled-components';

// HELPERS
import { fluidTypeVals, renderWithTheme } from '../../../test';
import { colours } from '../../helpers/colours';

// COMPONENT
import P from './index';

test('Render default P tag and styles', async () => {
  // Variables
  const minFS = 14;
  const maxFS = 17;
  const minLH = 17;
  const maxLH = 29;
  // Render
  const { getByTestId, container } = renderWithTheme(
    <P data-testid='test-text'>A tasty treat</P>
  );
  const el = getByTestId('test-text');
  const fontSizes = fluidTypeVals({ minLH, maxLH, maxFS, minFS });
  // Assert
  expect(el).toHaveTextContent('A tasty treat');
  expect(el).toHaveStyleRule('color', colours.charcoal);
  expect(el).toHaveStyleRule('margin', '0 0 20px');
  expect(el).toHaveStyleRule('padding', '0');
  expect(el).toHaveStyleRule('font-size', `${ minFS }px`);
  expect(el).toHaveStyleRule('font-size', fontSizes.minFont, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('font-size', `${ maxFS }px`, { media: 'screen and (min-width:1200px)' });
  expect(el).toHaveStyleRule('line-height', `${ minLH }px`);
  expect(el).toHaveStyleRule('line-height', fontSizes.minLineHeight, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('line-height', `${ maxLH }px`, { media: 'screen and (min-width:1200px)' });
  expect(container.firstChild).toMatchSnapshot();
});

test('Render P tag with core props', async () => {
  // Render
  const { getByTestId, container } = renderWithTheme(
    <P
      data-testid='test-text'
      fontSize='30px'
      color='#333'
      lineHeight='40px'
      align='right'
      margin='0 0 10px'
      padding='10px'
    >
      A tasty treat
    </P>
  );
  const el = getByTestId('test-text');
  // Assert
  expect(el).toHaveStyleRule('margin', '0 0 10px');
  expect(el).toHaveStyleRule('padding', '10px');
  expect(el).toHaveStyleRule('font-size', '30px');
  expect(el).toHaveStyleRule('line-height', '40px');
  expect(el).toHaveStyleRule('text-align', 'right');
  expect(el).toHaveStyleRule('color', '#333');
  expect(container.firstChild).toMatchSnapshot();
});

test('Render P tag with fluid type props', async () => {
  // Variables
  const minFS = 10;
  const maxFS = 18;
  const minLH = 18;
  const maxLH = 26;
  // Render
  const { getByTestId, container } = renderWithTheme(
    <P
      data-testid='test-text'
      minFont={ minFS }
      maxFont={ maxFS }
      minLine={ minLH }
      maxLine={ maxLH }
    >
      A tasty treat
    </P>
  );
  const el = getByTestId('test-text');
  // Assert
  const fontSizes = fluidTypeVals({ minLH, maxLH, maxFS, minFS });
  expect(el).toHaveStyleRule('font-size', `${ minFS }px`);
  expect(el).toHaveStyleRule('font-size', fontSizes.minFont, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('font-size', `${ maxFS }px`, { media: 'screen and (min-width:1200px)' });
  expect(el).toHaveStyleRule('line-height', `${ minLH }px`);
  expect(el).toHaveStyleRule('line-height', fontSizes.minLineHeight, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('line-height', `${ maxLH }px`, { media: 'screen and (min-width:1200px)' });
  expect(container.firstChild).toMatchSnapshot();
});