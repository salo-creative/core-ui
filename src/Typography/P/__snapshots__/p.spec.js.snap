// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render P tag with core props 1`] = `
.c0 {
  font-weight: 400;
  width: 100%;
  color: #333;
  text-align: right;
  margin: 0 0 10px;
  padding: 10px;
  font-size: 30px;
  line-height: 40px;
}

.c0 em,
.c0 i {
  font-style: italic;
}

.c0 b,
.c0 strong {
  font-weight: bold;
}

.c0 a {
  font-size: 1em;
  color: currentColor;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  font-weight: 600;
}

.c0 a:hover {
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  color: #8f8f8f;
}

.c0 pre {
  font-family: "Courier 10 Pitch",Courier,monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
}

.c0 code {
  font-family: Monaco,Consolas,"Andale Mono","DejaVu Sans Mono",monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
  background: #faf8f0;
}

<p
  class="c0"
  color="#333"
  data-testid="test-text"
  font-size="30px"
>
  A tasty treat
</p>
`;

exports[`Render P tag with fluid type props 1`] = `
.c0 {
  font-weight: 400;
  width: 100%;
  color: #262729;
  text-align: left;
  margin: 0 0 20px;
  padding: 0;
  font-size: 10px;
  line-height: 18px;
}

.c0 em,
.c0 i {
  font-style: italic;
}

.c0 b,
.c0 strong {
  font-weight: bold;
}

.c0 a {
  font-size: 1em;
  color: currentColor;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  font-weight: 600;
}

.c0 a:hover {
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  color: #8f8f8f;
}

.c0 pre {
  font-family: "Courier 10 Pitch",Courier,monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
}

.c0 code {
  font-family: Monaco,Consolas,"Andale Mono","DejaVu Sans Mono",monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
  background: #faf8f0;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: calc(10px + (18 - 10) * (100vw - 400px) / (1200 - 400));
    line-height: calc(18px + (26 - 18) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    font-size: 18px;
    line-height: 26px;
  }
}

<p
  class="c0"
  color="#262729"
  data-testid="test-text"
  font-size=""
>
  A tasty treat
</p>
`;

exports[`Render default P tag and styles 1`] = `
.c0 {
  font-weight: 400;
  width: 100%;
  color: #262729;
  text-align: left;
  margin: 0 0 20px;
  padding: 0;
  font-size: 14px;
  line-height: 17px;
}

.c0 em,
.c0 i {
  font-style: italic;
}

.c0 b,
.c0 strong {
  font-weight: bold;
}

.c0 a {
  font-size: 1em;
  color: currentColor;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  font-weight: 600;
}

.c0 a:hover {
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  color: #8f8f8f;
}

.c0 pre {
  font-family: "Courier 10 Pitch",Courier,monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
}

.c0 code {
  font-family: Monaco,Consolas,"Andale Mono","DejaVu Sans Mono",monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
  background: #faf8f0;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: calc(14px + (17 - 14) * (100vw - 400px) / (1200 - 400));
    line-height: calc(17px + (29 - 17) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    font-size: 17px;
    line-height: 29px;
  }
}

<p
  class="c0"
  color="#262729"
  data-testid="test-text"
  font-size=""
>
  A tasty treat
</p>
`;
